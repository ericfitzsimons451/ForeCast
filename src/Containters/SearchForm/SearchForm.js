import React, { Component } from 'react'
import './SearchForm.scss'
import { PropTypes } from 'prop-types'
import { weatherApiKey } from '../../Helpers/weatherApiKey'
import { Link } from 'react-router-dom'
import { getCityWeather } from '../../Thunks/getCityWeather'
import { connect } from 'react-redux'

export class SearchForm extends Component {
	constructor() {
		super()
		this.state = {
			searchInfo: '',
		}
	}

	handleChange = (e) => {
		const { name, value } = e.target
		this.setState({ [name]: value })
	}

	handleGetCityWeather = (e) => {
		const cityName = this.state.searchInfo
		const url = `https://api.weatherbit.io/v2.0/current?city=${cityName}&key=${weatherApiKey}`
		this.props.getCityWeather(url)
		this.setState({ searchInfo: '' })
	}

	render() {
		return (
			<form className='search'>
				<div className='search-box'>
					<input
						onChange={this.handleChange}
						name='searchInfo' value={this.state.searchInfo}
						placeholder='Search any US city'
						className='input'
					/>
					<Link
						to={`/cities/${this.state.searchInfo}`}
						onClick={this.handleGetCityWeather}
						className='submit-btn'>Get Current Weather
					</Link>
				</div>
			</form>
		)
	}
}

export const mapDispatchToProps = dispatch => ({
	getCityWeather: (url) => dispatch(getCityWeather(url))
})

SearchForm.propTypes = {
	getCityWeather: PropTypes.func
}


export default connect(null, mapDispatchToProps)(SearchForm)