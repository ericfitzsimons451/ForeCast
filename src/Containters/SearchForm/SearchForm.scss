.search {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.input {
    width: 25vw;
}