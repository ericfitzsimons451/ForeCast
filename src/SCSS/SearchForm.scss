.search {
	display: flex;
	flex-direction: column;
	justify-content: flex-start;
	align-items: center;
	background: url('https://www.chubmagazine.com/wp-content/uploads/2017/10/Greengrass.jpg');
	min-height: 100vh;
	min-width: 100vw;
}

.search-box {
	display: flex;
	flex-direction: column;
	justify-content: flex-start;
	align-items: center;
	background: rgba(169, 176, 168, 0.815);
	width: 50vw;
	height: 45vh;
	margin-top: 30px;
	opacity: 1;
	border-radius: 15px;
}

.input {
	width: 30vw;
	height: 8vh;
	font-size: 1.2rem;
	border-radius: 10px;
	margin: 40px;
}

.link-tag {
	border: 1px solid black;
	border-radius: 15px;
	width: 30vw;
	height: 50px;
	font-size: 1.8rem;
	padding: 20px;
	text-decoration: none;
}