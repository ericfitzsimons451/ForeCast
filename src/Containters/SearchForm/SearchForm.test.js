import React from 'react'
import { SearchForm } from './SearchForm'
import { shallow } from 'enzyme'

describe('searchForm', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<SearchForm getCityWeather={jest.fn}/>)
    })

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    describe('handleChange', () => {
        it('should update state with the name and value of the e.target', () => {
            let wrapper = shallow(<SearchForm />)
            const mockEvent = {
                target: {
                    searchInfo: '',
                }
            }
            const initialState = {
                searchInfo: '',
            }
            const updatedState = {
                searchInfo: '',
            }
            expect(wrapper.state()).toEqual(initialState)
            wrapper.instance().handleChange(mockEvent)
            expect(wrapper.state()).toEqual(updatedState)
        })
    })

    // describe('handleSubmit', () => {
    //     it('updates calls a props function', () => {
    //         wrapper = shallow(<SearchForm getCityWeather={jest.fn}/>)
    //         const mockEvent = { preventDefault: jest.fn() }
    //         const mockState = {
    //             searchInfo: 'Raleigh'
    //         }
    //         wrapper.instance().props.getCityWeather(mockState)
    //     })
    // })
})