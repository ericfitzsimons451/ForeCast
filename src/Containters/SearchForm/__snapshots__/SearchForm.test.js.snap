// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchForm should match the snapshot 1`] = `
<form
  className="search"
>
  <div
    className="search-box"
  >
    <input
      className="input"
      name="searchInfo"
      onChange={[Function]}
      placeholder="Search any US city"
      value=""
    />
    <Link
      className="submit-btn"
      onClick={[Function]}
      to="/cities/"
    >
      Get Current Weather
    </Link>
  </div>
</form>
`;
